src/extensions/mvs/tree/mvs/mvs-tree.ts

```diff
@@ -251,10 +296,27 @@ export const MVSTreeSchema = TreeSchema({
                 background_color: RequiredField(ColorT, 'Color of the canvas background. Can be either an X11 color name (e.g. `"red"`) or a hexadecimal code (e.g. `"#FF0011"`).'),
             },
         },
+        primitives: {
+            description: 'This node groups a list of geometrical primitives',
+            parent: ['structure', 'root'],
+            params: {
+                default_color: OptionalField(nullable(ColorT)),
+                default_label_color: OptionalField(nullable(ColorT)),
+                default_tooltip: OptionalField(nullable(str)),
+                transparency: OptionalField(nullable(float)),
+            },
+        },
+        primitive: {
+            description: 'This node represents a geometrical primitive',
+            parent: ['primitives'],
+            params: {
+                // TODO: proper validation
+                _union_: RequiredField(PrimitiveParams),
+            },
```

Comment:
@midlik I have reworked how the primitives are represented (kind is part of params with single `primitive` node type) and I would like to have support for union type here. Any idea how to do this nicely?

Reply:
Btw this will be useful for structure representation custom props as well.